### PR TITLE
Trader Joe migration and add V2.1 trades on BNB Chain

### PR DESCRIPTION
**Pull request by hsrvc**
models/trader_joe/avalanche_c/trader_joe_avalanche_c_schema.yml
models/trader_joe/avalanche_c/trader_joe_avalanche_c_trades.sql
models/trader_joe/avalanche_c/trader_joe_avalanche_c_trades_legacy.sql
models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades.sql
models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades_legacy.sql
models/trader_joe/avalanche_c/trader_joe_v2_avalanche_c_trades.sql
models/trader_joe/avalanche_c/trader_joe_v2_avalanche_c_trades_legacy.sql
models/trader_joe/bnb/trader_joe_bnb_schema.yml
models/trader_joe/bnb/trader_joe_bnb_sources.yml
models/trader_joe/bnb/trader_joe_bnb_trades.sql
models/trader_joe/bnb/trader_joe_bnb_trades_legacy.sql
models/trader_joe/bnb/trader_joe_v2_1_bnb_trades.sql
models/trader_joe/bnb/trader_joe_v2_1_bnb_trades_legacy.sql
models/trader_joe/bnb/trader_joe_v2_bnb_trades.sql
models/trader_joe/bnb/trader_joe_v2_bnb_trades_legacy.sql
models/trader_joe/trader_joe_schema.yml
models/trader_joe/trader_joe_trades.sql
models/trader_joe/trader_joe_trades_legacy.sql
seeds/dex/trades/dex_trades_seed.csv

### CODE REVIEW

**Comment by hsrvc** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by hsrvc** [comment]
recheck

**Comment by hsrvc** [comment]
kindly cc @Hosuke @jeff-dude 🙏

File: models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades_legacy.sql
Comment (Hosuke) [inline models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades_legacy.sql:1]: For the old/existed legacy spells, I am not sure we need to make them as select 1 dummy tables. We may leave them as they are.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @hsrvc 
- Partition by `block_month` ✅
- `trace_address` is removed ✅


Also all spells migrated in this PR their legacy models are changed to select 1.

**Comment by jeff-dude** [comment]
can you resolve merge conflicts, now that #4022 is merged into main?

**Comment by hsrvc** [comment]
> can you resolve merge conflicts, now that #4022 is merged into main?

done 🙏

File: models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades_legacy.sql
Comment (hsrvc) [reply models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades_legacy.sql:1]: Got it, I've changed existing spell's legacy tables back to as-is . Thanks 🙏

Comment (jeff-dude) [reply models/trader_joe/avalanche_c/trader_joe_v1_avalanche_c_trades_legacy.sql:1]: think we're good now!

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**